storage/segment_query.go

```diff
@@ -324,9 +324,10 @@ func collectTopNRawResultDocIDOrderByValues(
 	// Sort the result heap in place, and when done the items are sorted from left to
 	// in the right order based on the query sorting criteria (i.e., if the sort order
 	// is ascending, the leftmost item is the smallest item).
-	for results.Len() > 0 {
-		results.Pop()
-	}
+	// TODO(wjang): This just pops every item off the heap and returns an empty slice.
```

Comment:
As discussed offline, this should be done in a `Sort` API of the raw results heap. You can reassign this to me.